enumerables.rb

```diff
@@ -0,0 +1,160 @@
+module Enumerable
+
+  # my_each
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_each
+    return to_enum(:my_each) unless block_given?
+      i = 0
+      while i < self.size do
+        yield(self.to_a[i])
+        i += 1
+      end
+  end
+  
+  # my_each_with_index
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+    i = 0
+    while i < self.length do
+      yield(i,self[i])
+      i += 1
+    end
+  end
+
+  # my_select
+  
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_select
+    return to_enum(:my_select) unless block_given?
+    arr = []
+    i = 0
+    while i < self.length do
+      arr << self[i] if yield(self[i])
+      i += 1
+    end
+    arr
+  end
+  
+ # my all
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all?
+    unless block_given?
+      i = 0
+      while i < self.length do
+        return false unless self[i]
+        i += 1
+      end
+      return true
+    end
+    bool = true
+    i = 0
+    while i < self.length do
+      bool = false unless yield(self[i]) 
+      i += 1
+    end
+    bool
+  end
```

Comment:
`my_all?`, including `my_any?` and `my_none?` should all accept arguments, these argument can be a `Class`, `Regex` or `Integer`, `String` and so on. See details here https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-all-3F

Currently, there is `ArgumentError` when trying out with examples in the ruby doc:

```ruby
[1, 2i, 3.14].all?(Numeric)  
# => true
irb(main):003:0> [1, 2i, 3.14].my_all?(Numeric) 
# ArgumentError (wrong number of arguments (given 1, expected 0))
```
Please see other eaxmpls in the link provided above.

I would recommend add a dummy argument, and handle the method differently when the argument is not `nil`.
Also investigate the use of `===` as this will help with checking collection members, an example is shown below:
```ruby
irb(main):001:0> Integer === 2
=> true
irb(main):002:0> Integer === 2.3
=> false
irb(main):003:0> Numeric === 2i
=> true
irb(main):004:0> Integer === 3.5
=> false
irb(main):005:0> Float === 2.2
=> true
irb(main):006:0> /d/ === 'dog'
=> true
```


> As your code will be a bit more complex disable Style/CaseEquality cop, learn how to do it here: https://rubydoc.info/gems/rubocop/0.24.1/frames#Disabling_Cops_within_Source_Code


#### Do this for all other similar methods (`my_any?` and `my_none?`)